frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.unit.spec.tsx

```diff
@@ -0,0 +1,167 @@
+import fetchMock from "fetch-mock";
+import {
+  getBrokenUpTextMatcher,
+  renderWithProviders,
+  screen,
+} from "__support__/ui";
+import { testDataset } from "__support__/testDataset";
+import {
+  setupActionsEndpoints,
+  setupDatabasesEndpoints,
+  setupTableEndpoints,
+} from "__support__/server-mocks";
+import {
+  createMockQueryBuilderState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { createMockCard, createMockDataset } from "metabase-types/api/mocks";
+import {
+  PRODUCTS_ID,
+  SAMPLE_DB_ID,
+  createOrdersTable,
+  createProductsTable,
+  createReviewsTable,
+} from "metabase-types/api/mocks/presets";
+import { createMockEntitiesState } from "__support__/store";
+import type { Field } from "metabase-types/api";
+
+import ObjectDetail from "./ObjectDetail";
+
+const PRODUCTS_TABLE = createProductsTable();
+const ORDERS_TABLE = createOrdersTable();
+const HIDDEN_ORDERS_TABLE = createOrdersTable({
+  visibility_type: "hidden",
+});
+const REVIEWS_TABLE = createReviewsTable();
+
+interface SetupOpts {
+  hideOrdersTable?: boolean;
+}
+
+function setup({ hideOrdersTable = false }: SetupOpts = {}) {
```

Comment:
All the following API mocks + state mocks are inserted one by one until the test starts rendering, so I'm pretty they're needed. It might not be apparent from the get-go why all of these are required.